### PR TITLE
Respect HTML colors in labels

### PR DESCRIPTION
**Pull request by nyalldawson**
When enabled, this option will treat label contents as HTML, and
any FOREGROUND COLOR RELATED html formatting options will be
respected in the rendered label.

I repeat: ONLY HTML COLOR TAGS ARE RESPECTED. This is NOT a bug,
rather this feature has been designed as a "test of the waters"
with HTML formatting in labels, and accordingly only formatting
options which do not alter the font shape were considered.

On the plus side, it works correctly with ALL other label settings,
including shadows, buffers, curved labels, etc!

Sponsored by geoProRegio AG (www.geoproregio.ch)

Now some pretty pictures:

image

image

### CODE REVIEW

File: src/core/qgsrendercontext.cpp [hunk 1]
```diff
@@ -98,7 +98,8 @@ QgsRenderContext &QgsRenderContext::operator=( const QgsRenderContext &rh )
   mHasRenderedFeatureHandlers = rh.mHasRenderedFeatureHandlers;
   mCustomRenderingFlags = rh.mCustomRenderingFlags;
   setIsTemporal( rh.isTemporal() );
-  setTemporalRange( rh.temporalRange() );
+  if ( rh.isTemporal() )
```
Comment (m-kuhn) [inline src/core/qgsrendercontext.cpp:101]: Unrelated change?

**Comment by nyalldawson** [comment]
Now also supports underline, strikethrough, and overline:

image

**Comment by NathanW2** [comment]
;) hello html labels my old friend. Good to see you are back from the dead

**Review by m-kuhn — APPROVED** [review]
I have reviewed this with respect to

 - Basic code style sanity checks
 - Feature awesomeness
 - API looking reasonable
 - Tests are in place

And can give a +1 for all of that.

**Comment by nyalldawson** [comment]
Ok, this is ready for merge (if CI is happy)

**Comment by m-kuhn** [comment]
Thanks for the test work

**Comment by timlinux** [comment]
I wonder more generally what the use case of this is - are there any maps in the wild you can show that use mixed colours etc in the labels?

Just curious...more options are always nice :-)

**Comment by andreasneumann** [comment]
> I wonder more generally what the use case of this is - are there any maps in the wild you can show that use mixed colours etc in the labels?
> 
> Just curious...more options are always nice :-)

Yes - consider maps that show the state of industrial devices or infrastructure - e.g. waste-water structures with pump, valves, different levels. Color and bold can express the state of different sensors.
